src/future_driver.rs

```diff
@@ -0,0 +1,11 @@
+use futures::task::Poll;
+use futures::Future;
+use std::cell::RefCell;
+use std::rc::Rc;
+
+pub fn tick<F: Future>(f: Rc<RefCell<F>>) -> Poll<F::Output> {
+    let waker = futures_util::task::noop_waker();
+    // TODO: Is this safe?
```

Comment:
no, I think you need `F: Future+Unpin`.

Reply:
`Pin<T>` means that until the `T` is dropped it isn't ever moved -- crucially _not_ until the `Pin<T>` is dropped- once pinned, always pinned. And you can't control that here- depending on the other owners of the `Rc<_>`.